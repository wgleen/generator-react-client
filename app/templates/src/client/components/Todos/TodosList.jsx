import React, { Component } from 'react'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { getTodos } from '../../actions/todosActions'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'

class TodosList extends Component {
  componentWillMount () {
    const {
      getTodos
    } = this.props

    getTodos()
  }

  render () {
    const {
      todos
    } = this.props

    const _todos = todos.map((todo, i) => (
      <TableRow key={i}>
        <TableCell>{todo.id}</TableCell>
        <TableCell>{todo.title}</TableCell>
      </TableRow>
    ))

    return (
      <div>
        <h2>Todo list</h2>

        <Table>
          <TableHead>
            <TableRow>
              <TableCell>ID</TableCell>
              <TableCell>Title</TableCell>
            </TableRow>
          </TableHead>
          <TableBody>
            {_todos}
          </TableBody>
        </Table>
      </div>
    )
  }
}

const mapStateToProps = state => ({
  todos: state.todosReducer.todos.content
})

const mapDispatchToProps = dispatch =>
  bindActionCreators({ getTodos }, dispatch)

export default connect(mapStateToProps, mapDispatchToProps)(TodosList)