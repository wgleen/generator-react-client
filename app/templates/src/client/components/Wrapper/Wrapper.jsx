import React from 'react'
import { Provider } from 'react-redux'
import {
  store
} from '../../lib/store'

const Wrapper = props => (
  <Provider store={store}>
    {props.children}
  </Provider>
)

export default Wrapper