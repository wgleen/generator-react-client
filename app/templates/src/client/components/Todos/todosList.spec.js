import React from 'react'
import { mount } from 'enzyme'
import { random } from 'lodash'
import Wrapper from '../Wrapper/Wrapper'
import TodosList from './TodosList'

describe('<TodosList />', () => {
  it('should render correctly', () => {
    const component = mount(
      <Wrapper>
        <TodosList />
      </Wrapper>
    )

    expect(component).toMatchSnapshot()

    component.unmount()
  })

  it('should contain a h2 tag with "Todo list" text', () => {
    const component = mount(
      <Wrapper>
        <TodosList />
      </Wrapper>
    )

    expect(
      component
        .find('h2')
        .text()
    ).toEqual('Todo list')

    component.unmount()
  })

  it('should contain a Table with TableCells with "ID" and "Title" texts', () => {
    const component = mount(
      <Wrapper>
        <TodosList />
      </Wrapper>
    )

    const table = component.find('TableHead').find('TableCell')

    expect(table.at(0).text()).toEqual('ID')
    expect(table.at(1).text()).toEqual('Title')

    component.unmount()
  })

  it('should have a list of todos into TableRows', () => {
    const component = mount(
      <Wrapper>
        <TodosList />
      </Wrapper>
    )

    const _component = component.find('TodosList')
    const list = _component.props().todos
    const index = random(0, list.length - 1)
    const item = list[index]
    const row = component
      .find('TableBody')
      .find('TableRow').at(index)
    const cell = row.find('TableCell')

    expect(cell.at(0).text()).toEqual(String(item.id))
    expect(cell.at(1).text()).toEqual(item.title)

    component.unmount()
  })
})
