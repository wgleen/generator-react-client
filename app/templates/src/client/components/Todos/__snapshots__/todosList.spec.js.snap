// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodosList /> should render correctly 1`] = `
<Wrapper>
  <Provider
    store={
      Object {
        "dispatch": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
        Symbol(observable): [Function],
      }
    }
  >
    <Connect(TodosList)>
      <TodosList
        getTodos={[Function]}
        received={true}
        todos={
          Array [
            Object {
              "id": 0,
              "title": "Internal Configuration Technician",
            },
            Object {
              "id": 1,
              "title": "Chief Web Engineer",
            },
            Object {
              "id": 2,
              "title": "Global Functionality Planner",
            },
            Object {
              "id": 3,
              "title": "National Group Facilitator",
            },
            Object {
              "id": 4,
              "title": "Corporate Branding Analyst",
            },
            Object {
              "id": 5,
              "title": "Global Response Designer",
            },
          ]
        }
      >
        <div>
          <h2>
            Todo list
          </h2>
          <WithStyles(Table)>
            <Table
              classes={
                Object {
                  "root": "MuiTable-root-1",
                }
              }
              component="table"
              padding="default"
            >
              <table
                className="MuiTable-root-1"
              >
                <WithStyles(TableHead)>
                  <TableHead
                    classes={
                      Object {
                        "root": "MuiTableHead-root-2",
                      }
                    }
                    component="thead"
                  >
                    <thead
                      className="MuiTableHead-root-2"
                    >
                      <WithStyles(TableRow)>
                        <TableRow
                          classes={
                            Object {
                              "footer": "MuiTableRow-footer-7",
                              "head": "MuiTableRow-head-6",
                              "hover": "MuiTableRow-hover-5",
                              "root": "MuiTableRow-root-3",
                              "selected": "MuiTableRow-selected-4",
                            }
                          }
                          component="tr"
                          hover={false}
                          selected={false}
                        >
                          <tr
                            className="MuiTableRow-root-3 MuiTableRow-head-6"
                          >
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <th
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-head-9"
                                  scope="col"
                                >
                                  ID
                                </th>
                              </TableCell>
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <th
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-head-9"
                                  scope="col"
                                >
                                  Title
                                </th>
                              </TableCell>
                            </WithStyles(TableCell)>
                          </tr>
                        </TableRow>
                      </WithStyles(TableRow)>
                    </thead>
                  </TableHead>
                </WithStyles(TableHead)>
                <WithStyles(TableBody)>
                  <TableBody
                    classes={
                      Object {
                        "root": "MuiTableBody-root-16",
                      }
                    }
                    component="tbody"
                  >
                    <tbody
                      className="MuiTableBody-root-16"
                    >
                      <WithStyles(TableRow)
                        key="0"
                      >
                        <TableRow
                          classes={
                            Object {
                              "footer": "MuiTableRow-footer-7",
                              "head": "MuiTableRow-head-6",
                              "hover": "MuiTableRow-hover-5",
                              "root": "MuiTableRow-root-3",
                              "selected": "MuiTableRow-selected-4",
                            }
                          }
                          component="tr"
                          hover={false}
                          selected={false}
                        >
                          <tr
                            className="MuiTableRow-root-3"
                          >
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  0
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  Internal Configuration Technician
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                          </tr>
                        </TableRow>
                      </WithStyles(TableRow)>
                      <WithStyles(TableRow)
                        key="1"
                      >
                        <TableRow
                          classes={
                            Object {
                              "footer": "MuiTableRow-footer-7",
                              "head": "MuiTableRow-head-6",
                              "hover": "MuiTableRow-hover-5",
                              "root": "MuiTableRow-root-3",
                              "selected": "MuiTableRow-selected-4",
                            }
                          }
                          component="tr"
                          hover={false}
                          selected={false}
                        >
                          <tr
                            className="MuiTableRow-root-3"
                          >
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  1
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  Chief Web Engineer
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                          </tr>
                        </TableRow>
                      </WithStyles(TableRow)>
                      <WithStyles(TableRow)
                        key="2"
                      >
                        <TableRow
                          classes={
                            Object {
                              "footer": "MuiTableRow-footer-7",
                              "head": "MuiTableRow-head-6",
                              "hover": "MuiTableRow-hover-5",
                              "root": "MuiTableRow-root-3",
                              "selected": "MuiTableRow-selected-4",
                            }
                          }
                          component="tr"
                          hover={false}
                          selected={false}
                        >
                          <tr
                            className="MuiTableRow-root-3"
                          >
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  2
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  Global Functionality Planner
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                          </tr>
                        </TableRow>
                      </WithStyles(TableRow)>
                      <WithStyles(TableRow)
                        key="3"
                      >
                        <TableRow
                          classes={
                            Object {
                              "footer": "MuiTableRow-footer-7",
                              "head": "MuiTableRow-head-6",
                              "hover": "MuiTableRow-hover-5",
                              "root": "MuiTableRow-root-3",
                              "selected": "MuiTableRow-selected-4",
                            }
                          }
                          component="tr"
                          hover={false}
                          selected={false}
                        >
                          <tr
                            className="MuiTableRow-root-3"
                          >
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  3
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  National Group Facilitator
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                          </tr>
                        </TableRow>
                      </WithStyles(TableRow)>
                      <WithStyles(TableRow)
                        key="4"
                      >
                        <TableRow
                          classes={
                            Object {
                              "footer": "MuiTableRow-footer-7",
                              "head": "MuiTableRow-head-6",
                              "hover": "MuiTableRow-hover-5",
                              "root": "MuiTableRow-root-3",
                              "selected": "MuiTableRow-selected-4",
                            }
                          }
                          component="tr"
                          hover={false}
                          selected={false}
                        >
                          <tr
                            className="MuiTableRow-root-3"
                          >
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  4
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  Corporate Branding Analyst
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                          </tr>
                        </TableRow>
                      </WithStyles(TableRow)>
                      <WithStyles(TableRow)
                        key="5"
                      >
                        <TableRow
                          classes={
                            Object {
                              "footer": "MuiTableRow-footer-7",
                              "head": "MuiTableRow-head-6",
                              "hover": "MuiTableRow-hover-5",
                              "root": "MuiTableRow-root-3",
                              "selected": "MuiTableRow-selected-4",
                            }
                          }
                          component="tr"
                          hover={false}
                          selected={false}
                        >
                          <tr
                            className="MuiTableRow-root-3"
                          >
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  5
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                            <WithStyles(TableCell)>
                              <TableCell
                                classes={
                                  Object {
                                    "body": "MuiTableCell-body-10",
                                    "footer": "MuiTableCell-footer-11",
                                    "head": "MuiTableCell-head-9",
                                    "numeric": "MuiTableCell-numeric-12",
                                    "paddingCheckbox": "MuiTableCell-paddingCheckbox-14",
                                    "paddingDense": "MuiTableCell-paddingDense-13",
                                    "paddingNone": "MuiTableCell-paddingNone-15",
                                    "root": "MuiTableCell-root-8",
                                  }
                                }
                                numeric={false}
                              >
                                <td
                                  aria-sort={null}
                                  className="MuiTableCell-root-8 MuiTableCell-body-10"
                                >
                                  Global Response Designer
                                </td>
                              </TableCell>
                            </WithStyles(TableCell)>
                          </tr>
                        </TableRow>
                      </WithStyles(TableRow)>
                    </tbody>
                  </TableBody>
                </WithStyles(TableBody)>
              </table>
            </Table>
          </WithStyles(Table)>
        </div>
      </TodosList>
    </Connect(TodosList)>
  </Provider>
</Wrapper>
`;
